internal/cloudapi/v2/handler.go

```diff
@@ -314,6 +315,30 @@ func targetResultToUploadStatus(t *target.TargetResult) (*UploadStatus, error) {
 	return us, nil
 }
 
+// GetComposeList returns a list of the root job UUIDs
+func (h *apiHandlers) GetComposeList(ctx echo.Context) error {
+	jobs, err := h.server.workers.AllRootJobIDs()
+	if err != nil {
+		return HTTPErrorWithInternal(ErrorGettingComposeList, err)
+	}
+
+	// Gather up the details of each job
+	var stats []ComposeStatus
+	for _, jid := range jobs {
+		s, err := h.getJobIDComposeStatus(jid)
+		if err != nil {
+			// TODO log this error?
```

Comment:
Under what circumstance could this fail? Fwiw, I think we should log the errors but I do wonder under what circumstance AllRootJobIDs() and getJobIDComposeStatus(id) can get out of sync

Reply:
Since it is gathering the jobs as a separate step it is possible something could come along and delete them before it is done iterating the list. Maybe not in practice, especially now before adding a `delete` route but it seems remotely possible. I'm not sure it is worth logging though, there's nothing that can be done if it vanished, other than skipping it,

Reply:
This is another case of what would it do with the error? I don't want that to block listing other non-error jobs so it's better to ignore them and carry on.

Reply:
I almost agree :) But I think that would end up logging useless errors. I'll leave it as-is for now and we can revisit later if it ends up that a message could help.